planning/exact-types/requirements.md

```
@@ -0,0 +1,1360 @@
+A template literal type is a derived union over the strings produced by
+substituting each hole with a value from its hole type. Its exactness is the
+**conjunction** of the exactnesses of its hole types:
+
+- If every hole is an exact union over a finite set of literal types, the
+  resulting set of strings is finite and known, so the template literal
+  type is **exact**.
+- If any hole is `string`, `number`, an inexact union, or any other type
+  with an unknown tail, the resulting set is open, so the template literal
+  type is **inexact**.
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)